src/dbnode/storage/index/compaction/plan.go

```diff
@@ -0,0 +1,250 @@
+// Copyright (c) 2018 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package compaction
+
+import (
+	"errors"
+	"fmt"
+	"sort"
+	"time"
+
+	"github.com/m3db/m3/src/dbnode/storage/index/segments"
+)
+
+var (
+	errMutableCompactionAgeNegative = errors.New("mutable compaction age must be positive")
+	errLevelsUndefined              = errors.New("compaction levels are undefined")
+	errMaxImmutableCompactionSize   = errors.New("max immutable compaction size must be positive")
+)
+
+var (
+	// DefaultLevels are the default Level(s) used for compaction options.
+	DefaultLevels = []Level{ // i.e. tiers for compaction [0, 262K), [262K, 524K), [524K, 1M), [1M, 2M), [2M, 8M)
+		Level{
+			MinSizeInclusive: 0,
+			MaxSizeExclusive: 1 << 18,
+		},
+		Level{
+			MinSizeInclusive: 1 << 18,
+			MaxSizeExclusive: 1 << 19,
+		},
+		Level{
+			MinSizeInclusive: 1 << 19,
+			MaxSizeExclusive: 1 << 20,
+		},
+		Level{
+			MinSizeInclusive: 1 << 20,
+			MaxSizeExclusive: 1 << 21,
+		},
+		Level{
+			MinSizeInclusive: 1 << 21,
+			MaxSizeExclusive: 1 << 23,
+		},
+	}
+
+	// DefaultOptions are the default compaction PlannerOptions.
+	DefaultOptions = PlannerOptions{
+		MaxImmutableCompactionSize: 1 << 22,                            // ~4M
```

Comment:
Yeah could we remove this for now and add it back in when introducing different planner types?

Reply:
sure thing, done in next patch.

Reply:
done